src/components/CategoryCard/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react";
+import { OPEN__MODAL } from "../../constants";
+import { useQuizData } from "../../context/quizContext";
+import { Card, CardTitle, CardWrap, ErrorMessage, Heading, Main, StartButton } from "./styles";
+
+const categories = [
+	{
+		id: "1",
+		name: "Random",
+		type: "random",
+	},
+	{
+		id: "2",
+		name: "IPL",
+		type: "",
+	},
+];
+
+function CategoryCard() {
+	const [chosen, setChosen] = useState("");
+	const { dispatch } = useQuizData();
+	const [message, setMessage] = useState("");
+
+	const onClickHandler = (e: React.MouseEvent<HTMLDivElement>, item: string) => {
+		return setChosen(item);
+	};
+
+	const openModalHandler = () => {
+		if (chosen.length > 0) {
+			return dispatch({ type: OPEN__MODAL, payload: { modalType: "addName", data: chosen } });
+		}
+		return setMessage("Please click on a Category");
+	};
+
+	return (
+		<Main>
+			<Heading>Choose a Category</Heading>
+			<CardWrap>
+				{categories.map((el) => (
+					<Card
```

Comment:
You can destructure el here
`const {id, type, ...} = el;`

Don't use `el` and such namings while mapping objects, use meaningful names so that they are easy to understand
e.g. `category` here

Reply:
done